src/xcode/ENA/ENA/Source/Services/HealthCertificateValidation/RulesDownloadService.swift

```diff
@@ -18,257 +18,34 @@ class RulesDownloadService: RulesDownloadServiceProviding {
 	}
 	
 	// MARK: - Protocol RulesDownloadServiceProviding
 	
 	func downloadRules(
 		ruleType: HealthCertificateValidationRuleType,
-		completion: @escaping (Result<[Rule], HealthCertificateValidationError>) -> Void
+		completion: @escaping (Result<[Rule], DCCDownloadRulesError>) -> Void
 	) {
-		switch ruleType {
-		case .acceptance:
-			let eTag = store.acceptanceRulesCache?.lastValidationRulesETag
-			downloadValidationRules(eTag: eTag, ruleType: ruleType, completion: completion)
-			
-		case .invalidation:
-			let eTag = store.invalidationRulesCache?.lastValidationRulesETag
-			downloadValidationRules(eTag: eTag, ruleType: ruleType, completion: completion)
-			
-		case .boosterNotification:
-			let eTag = store.boosterRulesCache?.lastValidationRulesETag
-			downloadBoosterRules(eTag: eTag, ruleType: ruleType, completion: completion)
-		}
-	}
-	
-	// MARK: - Private
-
-	private let store: Store
-	private let client: Client
-	private let validationRulesAccess: ValidationRulesAccessing
-	private let signatureVerifier: SignatureVerification
-		
-	private func downloadValidationRules(
-		eTag: String?,
-		ruleType: HealthCertificateValidationRuleType,
-		completion: @escaping (Result<[Rule], HealthCertificateValidationError>) -> Void
-	) {
-		Log.info("Download validation rules.")
-		
-		client.getDCCRules(
-			eTag: eTag,
-			isFake: false,
-			ruleType: ruleType,
-			completion: { [weak self] result in
-				guard let self = self else {
-					Log.error("Could not create strong self", log: .vaccination)
-					completion(.failure(.RULE_CLIENT_ERROR(ruleType)))
-					return
-				}
-				
-				switch result {
-				case let .success(packageDownloadResponse):
-					self.rulesDownloadingSuccess(
-						ruleType: ruleType,
-						packageDownloadResponse: packageDownloadResponse,
-						completion: completion
-					)
-	
-				case let .failure(failure):
-					self.rulesDownloadingFailure(
-						ruleType: ruleType,
-						failure: failure,
-						completion: completion
-					)
-				}
-			}
-		)
-	}
-	
-	private func downloadBoosterRules(
-		eTag: String?,
-		ruleType: HealthCertificateValidationRuleType,
-		completion: @escaping (Result<[Rule], HealthCertificateValidationError>) -> Void
-	) {
-		Log.info("Download booster rules.")
-
-		client.getBoosterNotificationRules(
-			eTag: eTag, isFake: false,
-			completion: { [weak self] result in
-				guard let self = self else {
-					Log.error("Could not create strong self", log: .vaccination)
-					completion(.failure(.RULE_CLIENT_ERROR(.invalidation)))
+		let resource = DCCRulesResource(ruleType: ruleType)
+		restServiceProvider.load(resource) { result in
+			switch result {
+			case let .success(validationRulesModel):
+				completion(.success(validationRulesModel.rules))
+			case let .failure(error):
+				guard let customError = resource.customError(for: error) else {
+					Log.error("Unhandled error \(error.localizedDescription)", log: .vaccination)
 					return
```

Comment:
Why not calling completion with the error? Else the caller would wait for a completion and get stuck.

Reply:
👍